### DISCUSSION TITLE
Ledger Metrics

### DISCUSSION THREAD

**Discussion by yana-novikova** [XRPLBounties/Proposals#18] [Closed (claimed or cancelled bounty)]
### Discussion thread for bounty: [Ledger Metrics](https://github.com/XRPLBounties/Proposals/blob/main/bounties/0018%20Ledger%20Metrics.md) ([PR](https://github.com/XRPLBounties/Proposals/pull/21))

(Previous name: **Summarize individual ledgers in XRPL Explorer**)

**This bounty is currently closed.**

Details below.

**Comment by ckniffen** [comment]
@yana-novikova created #21

**Reply by intelliot** [reply]
Update: This bounty has been approved and is now available for developers to pick up. The original post has been updated with a link to the bounty details. Interested? Discuss by writing a comment below!

**Comment by TusharPardhe** [comment]
Current design that I am working on: 

![Screenshot 2022-09-03 at 9 59 43 PM](https://user-images.githubusercontent.com/31487192/188279836-617c4199-6d7e-4b06-9d29-cb1a7e125347.png)


Some clarifications required: 
1. Breakdown of transactions by type (DEX, Payments, Account, NFT, XChain) => Do we need to categorize each tx type like OfferCreate and OfferCancel as DEX txs ? Please elaborate. If yes, do we have a list or we need to manually bifurcate.
2. A special call out (icon) if there are no tecINVARIANT_FAILED transactions => What do you mean by a "call out icon" ?

**Reply by TusharPardhe** [reply]
Desktop:
![Screenshot 2022-09-04 at 12 08 54 PM](https://user-images.githubusercontent.com/31487192/188300881-5cbc7f47-f175-45b7-89f4-03d5d5117555.png)

Mobile:
![Screenshot 2022-09-04 at 12 29 17 PM](https://user-images.githubusercontent.com/31487192/188301502-3ab6a33a-11e2-4b58-8035-e58627606236.png)

First point has been resolved (Header names can be changed)

Resolution of point two `A special call out (icon) if there are no tecINVARIANT_FAILED transactions => What do you mean by a "call out icon" ?` still pending.

**Comment by ckniffen** [comment]
The types of transactions should be grouped for example all Offer ones OfferCreate, OfferCancel would be rolled up into one value like DEX.  I will work on creating a list.  As part of the refactor of transaction types in the explorer will will be adding enums to group transaction types into categories.

The goal of the second milestone was to have the object types show the total number on that ledger.  I do not know how to get those values.

**Reply by TusharPardhe** [reply]
Currently, XRPL "ledger" web socket method (https://xrpl.org/ledger.html#ledger) returns a field named "LedgerEntryType" which comes from this list: https://xrpl.org/ledger-object-types.html

I think this solves it.
PS: We can give them more meaningful UI names, which imo is a better solution. What do you think?

**Comment by TusharPardhe** [comment]
PR for adding ledger metrics: https://github.com/ripple/explorer/pull/286/files

@yana-novikova @intelliot @ckniffen

**Comment by golobitch** [comment]
Since there was no change on original PR, I have decided to implement this feature. PR is already open: https://github.com/ripple/explorer/pull/328

@intelliot @ckniffen 

I grouped transaction types per category as I saw fit. This can be changed by your specification and it is quite trivial.

**Comment by JST5000** [comment]
We've accepted both @golobitch and @TusharPardhe's submissions for the 1st milestone of this bounty. We decided to pay both in full for the unique contributions they gave. At this point no further submissions for milestone 1 are being considered. 

Milestone 2 is still open though!

**Reply by TusharPardhe** [reply]
@JST5000 Adding to your point, last time I discussed with @ckniffen for Milestone 2 we need design and a bit of clarity (detailed description) because there will be lots of transaction types.

**Reply by JST5000** [reply]
Ah, thanks for adding that - we should update here once we have that clarity :)

**Comment by golobitch** [comment]
1832

**Reply by JST5000** [reply]
Thanks!

**Comment by jonathanlei** [comment]
closing this bounty until further decisions are made regarding design and milestone 2.
